src/ext_slice.rs

```diff
@@ -648,6 +649,150 @@ pub trait ByteSlice: Sealed {
         self.as_bytes().ends_with(suffix.as_ref())
     }
 
+    /// Returns the index of the first occurrence of any of the bytes in the
+    /// provided set.
+    ///
+    /// The `byteset` may be any type that can be cheaply converted into a
+    /// `&[u8]`. This includes, but is not limited to, `&str` and `&[u8]`, but
+    /// note that passing a `&str` which contains multibyte characters may not
+    /// behave as you expect.
```

Comment:
I'd just add here for extra clarity, "... as you expect: each byte in the `&str` is treated as an individual member of the byte set."